native/dgesvd.go

```diff
@@ -0,0 +1,1241 @@
+					} else {
+						// Insufficient workspace for a fast algorithm.
+						itau := 0
+						iwork := itau + n
+
+						// Compute A = Q*R, copying result to U.
+						impl.Dgeqrf(m, n, a, lda, work[itau:], work[iwork:], lwork-iwork)
+						impl.Dlacpy(blas.Lower, m, n, a, lda, u, ldu)
+
+						// Generate Q in U.
+						impl.Dorgqr(m, m, n, u, ldu, work[itau:], work[iwork:], lwork-iwork)
+
+						// Copy R from A to VT, zeroing out below it.
+						impl.Dlacpy(blas.Upper, n, n, a, lda, vt, ldvt)
+						if n > 1 {
+							impl.Dlaset(blas.Lower, n-1, n-1, 0, 0, vt[ldvt:], ldvt)
+						}
+						ie := itau
+						itauq := ie + n
+						itaup := itauq + n
+						iwork = itaup + n
+
+						// Bidiagonalize R in VT.
+						impl.Dgebrd(n, n, vt, ldvt, s, work[ie:], work[itauq:], work[itaup:], work[iwork:], lwork-iwork)
+
+						// Multiply Q in U by left bidiagonalizing vectors in VT.
+						impl.Dormbr(lapack.ApplyQ, blas.Right, blas.NoTrans, m, n, n, vt, ldvt, work[itauq:], u, ldu, work[iwork:], lwork-iwork)
+
+						// Generate right bidiagonizing vectors in VT.
+						impl.Dorgbr(lapack.ApplyP, n, n, n, vt, ldvt, work[itaup:], work[iwork:], lwork-iwork)
+						iwork = ie + n
+
+						// Perform bidiagonal QR iteration, computing left singular
+						// vectors of A in U and computing right singular vectors
+						// of A in VT.
+						impl.Dbdsqr(blas.Upper, n, n, m, 0, s, work[ie:], vt, ldvt, u, ldu, dum, 1, work[iwork:])
+					}
+				}
+			}
+		} else {
+			// Path 10.
+			ie = 0
+			itauq := ie + n
+			itaup := itauq + n
+			iwork := itaup + n
+
+			// Bidiagonalize A
+			impl.Dgebrd(m, n, a, lda, s, work[ie:], work[itauq:], work[itaup:], work[iwork:], lwork-iwork)
+			if wantuas {
+				// Left singular vectors are desired in U. Copy result to U and
+				// generate left biadiagonalizing vectors in U.
+				impl.Dlacpy(blas.Lower, m, n, a, lda, u, ldu)
+				var ncu int
+				if wantus {
+					ncu = n
+				}
+				if wantua {
+					ncu = m
+				}
+				impl.Dorgbr(lapack.ApplyQ, m, ncu, n, u, ldu, work[itauq:], work[iwork:], lwork-iwork)
+			}
+			if wantvas {
+				// Right singular vectors are desired in VT. Copy result to VT and
+				// generate left biadiagonalizing vectors in VT.
+				impl.Dlacpy(blas.Upper, n, n, a, lda, vt, ldvt)
+				impl.Dorgbr(lapack.ApplyP, n, n, n, vt, ldvt, work[itaup:], work[iwork:], lwork-iwork)
+			}
+			if wantuo {
+				panic(noSVDO)
+			}
+			if wantvo {
+				panic(noSVDO)
+			}
+			iwork = ie + n
+			var nru, ncvt int
+			if wantuas || wantuo {
+				nru = m
+			}
+			if wantun {
+				nru = 0
+			}
+			if wantvas || wantvo {
+				ncvt = n
+			}
+			if wantvn {
+				ncvt = 0
+			}
+			if !wantuo && !wantvo {
+				// Perform bidiagonal QR iteration, if desired, computing left
+				// singular vectors in U and right singular vectors in VT.
+				ok = impl.Dbdsqr(blas.Upper, n, ncvt, nru, 0, s, work[ie:], vt, ldvt, u, ldu, dum, 1, work[iwork:])
+			} else {
```

Comment:
Either have both other branches here or leave a comment that this will be two branches when the complete implementation exists.